Lib/asyncio/selector_events.py

```diff
@@ -1065,16 +1077,61 @@ def write(self, data):
             self._loop._add_writer(self._sock_fd, self._write_ready)
 
         # Add it to the buffer.
-        self._buffer.extend(data)
+        self._buffer.append(data)
         self._maybe_pause_protocol()
 
-    def _write_ready(self):
+    def _get_sendmsg_buffer(self):
+        return itertools.islice(self._buffer, SC_IOV_MAX)
+
+    def _write_sendmsg(self):
         assert self._buffer, 'Data should not be empty'
+        if self._conn_lost:
+            return
+        try:
+            n = self._sock.sendmsg(self._get_sendmsg_buffer())
+            self._adjust_leftover_buffer(n)
```

Comment:
Naming nit: Maybe rename n -> nbytes? I had guessed that sendmsg returns the number of buffers processed, not the number of bytes.

Reply:
done